python/sparkdl/estimators/keras_image_file_estimator.py

```diff
@@ -0,0 +1,284 @@
+        if not self.isDefined(self.loss):
+            raise ValueError("Required Param loss must be specified but was not.")
+        if not kmutil.is_valid_loss_function(self.getLoss()):
+            raise ValueError("Named loss not supported in Keras: %s" % self.getLoss())
+        if not self.isDefined(self.inputCol):
+            raise ValueError("Input column must be defined")
+        if not self.isDefined(self.outputCol):
+            raise ValueError("Output column must be defined")
+        return True
+
+    def _getNumpyFeaturesAndLabels(self, dataset):
+        """
+        We assume the training data fits in memory on a single server.
+        The input dataframe is converted to numerical image features and
+        broadcast to all the worker nodes.
+        """
+        image_uri_col = self.getInputCol()
+        label_col = None
+        if self.isDefined(self.labelCol) and self.getLabelCol() != "":
+            label_col = self.getLabelCol()
+        tmp_image_col = self._loadedImageCol()
+        image_df = self.loadImagesInternal(dataset, image_uri_col).dropna(subset=[tmp_image_col])
+
+        localFeatures = []
+        localLabels = []
+        is_with_labels = label_col is not None
+
+        if is_with_labels:
+            label_schema = image_df.schema[label_col]
+            assert isinstance(label_schema.dataType, spla.VectorUDT), \
+                "must encode labels in one-hot vector format"
+
+        for row in image_df.collect():
+            spimg = row[tmp_image_col]
+            features = imageStructToArray(spimg)
+            localFeatures.append(features)
+            if is_with_labels:
+                try:
+                    _keras_label = row[label_col].array
+                except ValueError:
+                    raise ValueError("Cannot extract encoded label array")
+                localLabels.append(_keras_label)
+
+        if len(localFeatures) == 0:
+            raise ValueError("Given empty dataset!")
+
+        # We must reshape input data to form to the same size so as to be stacked
+        X = np.stack(localFeatures, axis=0)
+        y = None
+        if is_with_labels:
+            if not localLabels:
+                raise ValueError("Failed to load any labels from dataset, but labels are required")
+            y = np.stack(localLabels, axis=0)
+            assert y.shape[0] == X.shape[0], "must have same amount of features and labels"
+        return X, y
+
+    def _collectModels(self, kerasModelsBytesRDD):
+        """
+        Collect Keras models on workers to MLlib Models on the driver.
+        :param kerasModelBytesRDD: RDD of (param_map, model_bytes) tuples
+        :param paramMaps: list of ParamMaps matching the maps in `kerasModelsRDD`
+        :return: list of MLlib models
+        """
+        transformers = []
+        for (param_map, model_bytes) in kerasModelsBytesRDD.collect():
```

Comment:
that may totally blow up, but I guess this is ok for now...

Reply:
Added warning in `.fit` function (docs are only shown in public functions).